compiler/src/iree/compiler/InputConversion/StableHLO/StableHLOToIREEInputDialects.cpp

```diff
@@ -529,6 +529,8 @@ struct ConvertStableHloToIreeInputDialects final
           ml_program::GlobalLoadConstOp::getOperationName(),
           ml_program::GlobalStoreOp::getOperationName(),
           tensor::FromElementsOp::getOperationName(),
+          tensor::CollapseShapeOp::getOperationName(),
```

Comment:
Looks fine, but this seems a bit round about. Why not just use these as template parameter list?

Reply:
Not the original author but I believe this is so that we do not generate a separate implementation for each operation. This could likely be cleaned up by having separate invocations instead of a for loop.

Reply:
(ack that you arent the original author, but since you are here :) ).

Just why is this better than
```
patterns.add<func::ReturnOp, func::CallOp, cf::CondBranchOp, ....>(...)
```

Reply:
Reworked.